bin/hrx

```diff
@@ -67,28 +75,37 @@ class HRX::CLI < Thor
     end
 
     hrx = HRX::Archive.load(archive)
-    files_extracted = hrx.entries.count do |entry|
-      destination = File.join(directory, entry.path)
-      if entry.is_a?(HRX::Directory)
-        FileUtils.mkdir_p(destination)
-        false
-      else
-        FileUtils.mkdir_p(File.dirname(destination))
-        if !File.exist?(destination) ||
-           file_collision(destination) {entry.content}
-          File.write(destination, entry.content)
-          true
+    if options[:level]
+      if options[:level] < 1
+        raise Thor::Error.new("--level must be greater than or equal to 2.")
+      elsif !options[:level].is_a?(Integer)
+        raise Thor::Error.new("--level must be an integer.")
+      end
+
+      files_extracted = hrx.entries.group_by do |e|
+        dir = File.dirname(e.path)
+        dir == "." ? [] : dir.split("/")[0...options[:level]]
+      end.sum do |(components, entries)|
```

Comment:
`.sum` was added in Ruby 2.4, which would explain the Travis failure on Ruby 2.3

Reply:
Updated the required Ruby versions to make this available.